src/mjolnir/landmark_builder.cc

```diff
@@ -168,5 +221,44 @@ std::vector<Landmark> LandmarkDatabase::get_landmarks_in_bounding_box(const doub
   return landmarks;
 }
 
+bool BuildLandmarkFromPBF(const boost::property_tree::ptree& pt,
+                          const std::vector<std::string>& input_files) {
+  // parse config to get landmark database
+  auto db = pt.get_optional<std::string>("landmarks");
+  if (!db) {
+    LOG_ERROR("Landmarks config info not found. Landmarks builder will not be created.");
+    return false;
+  }
+
+  const filesystem::path parent_dir = filesystem::path(*db).parent_path();
+  if (!filesystem::exists(parent_dir) && !filesystem::create_directories(parent_dir)) {
+    LOG_ERROR("Can't create parent directory " + parent_dir.string());
+    return false;
+  }
```

Comment:
we should remove the database right here if it exists

Reply:
yeah i was trying to say before that if we open the database in non-read-only mode and its there we should just blow it away. that can happen inside the database construction imho, unless we see some future need to be able to append to the database?

Reply:
agree. for now it makes sense to allow writing only when a database is created. I'll update the code.